### PR TITLE
Added major and minor gridline option to muon guis

### LINKED ISSUE #29018: Grid line options in muon analysis
The scientists would like to be able to add major and minor gridlines to the muon analysis plots

### PR DESCRIPTION
**Pull request by tolu28-coder**
**Description of work.**
Added 2 buttons on plotting toolbar used in Muon analysis gui and Frequency domain analysis gui which 
allow users to show major or minor gridlines.

**To test:**
1. Open Muon analysis gui.
2. Select 1 to show major gridlines and 2 to show minor gridlines.
3. The same should hold true for Frequency domain analysis gui.
tempsnip

Fixes #29018 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_toolbar.py [hunk 1]
```diff
@@ -15,25 +16,79 @@
...
+            for ax in self.canvas.figure.get_axes():
+                ax.grid(True , color='black')
+            self.is_major_grid_on = True
+        self.canvas.draw()
+
+    def show_minor_gridlines(self):
+        if self.is_minor_grid_on:
+            for ax in self.canvas.figure.get_axes():
+                ax.minorticks_off()
+            self.is_major_grid_on = False
```
Comment (StephenSmith25) [inline scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_toolbar.py:88]: Should this be self.is_minor_grid_on? And likewise for the line below?

Comment (tolu28-coder) [reply scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_toolbar.py:88]: yes it should , thank you

**Review by StephenSmith25 — COMMENTED** [review]
Looks good, just suggesting a small rewording of the docs. Theres one small issue:

If you turn either of the grid lines on - switch to tiled plot and switch back - and then try switch the grid lines on again you'll have to press the button twice. This is probably due to not resetting the flag when we create a new canvas, which happens here https://github.com/mantidproject/mantid/blob/1bf2c77dd6dd4ca5e514a9b99b44cb46ad6ec882/scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py#L86

If we just reset the self.is_minor/major_grid_on flag when we create a new canvas it should work. Theres a few ways you could do this, for example add a method to the toolbar

```
def reset_gridline_flags()
```

Which we'd call when creating a new canvas.

File: docs/source/release/v5.1.0/muon.rst [hunk 2]
```diff
@@ -33,6 +33,8 @@ New Features
 - Updated rounding for time zero and first good data to be 3 decimal places.
 - Added double pulse analysis, see :ref:`Muon home tab <muon_home_tab-ref>`.
 - Added multi-period support to the LoadMuonNexusV2 algorithm.
+- Added two buttons to the Muon analysis and Frequency domain analysis plot toolbar to allow users to show major and minor gridlines.
+  domain analysis gui.
```
Comment (StephenSmith25) [inline docs/source/release/v5.1.0/muon.rst:37]: Just need to remove this extra line and then we're good to go.

**Review by StephenSmith25 — COMMENTED** [review]
Looks good to go, works well 👍 . Just have an extra line in the docs that should be removed

**Review by StephenSmith25 — DISMISSED** [review]
Looks good to go 👍

**Comment by gemmaguest** [comment]
@tolu28-coder Could I get you to squash the commits to keep the history tidy? It's not a big deal on this PR but good practice to get into. I'll give you a call to explain what I mean.

**Comment by gemmaguest** [comment]
@StephenSmith25 Could you re-approve this? The code hasn't changed (we've just squashed commits and rebased on master) so a cursory glance as a sanity check should be all that's needed.

**Review by StephenSmith25 — APPROVED** [review]
Looks good to me 👍

**Pull request merged**